src/Compilers/Core/Portable/Options/EditorConfigSyntaxTreeOptionsProvider.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Text;
+
+namespace Microsoft.CodeAnalysis.Options
+{
+    public sealed class EditorConfigSyntaxTreeOptionsProvider : SyntaxTreeOptionsProvider
+    {
+        public ImmutableArray<AdditionalText> EditorConfigFiles { get; }
+
+        public EditorConfigSyntaxTreeOptionsProvider(ImmutableArray<AdditionalText> editorConfigFiles)
+        {
+            EditorConfigFiles = editorConfigFiles.NullToEmpty();
+        }
+
+        public EditorConfigSyntaxTreeOptionsProvider AddEditorConfigFiles(IEnumerable<AdditionalText> editorConfigFiles)
+        {
+            return new EditorConfigSyntaxTreeOptionsProvider(this.EditorConfigFiles.AddRange(editorConfigFiles));
+        }
+
+        public EditorConfigSyntaxTreeOptionsProvider RemoveEditorConfigFiles(IEnumerable<AdditionalText> editorConfigFiles)
+        {
+            return new EditorConfigSyntaxTreeOptionsProvider(this.EditorConfigFiles.RemoveRange(editorConfigFiles));
+        }
+
+        public override OptionSet GetOptionsForSyntaxTreePath(string path)
+        {
+            // TODO: implement.
```

Comment:
In general for compiler features we avoid TODO in feature branches instead preferring PROTOTYPE.  Makes it easy to see of a merge of a feature to master is clean because we can grep for PROTOTYPE (should have 0).  Grepping for TODO is not useful.